scripts/bluetooth/mesh/mesh_dfu_metadata.py

```diff
@@ -0,0 +1,425 @@
+
+        with open(self.elf_path, 'rb') as elf_file:
+            elf = ELFFile(elf_file)
+            cdp0_value = self.__read_symbol_data(elf, addr)
+
+            # The format of the composition data is defined by `struct bt_mesh_comp` type.
+            # The format of an element is defined by `struct bt_mesh_elem` type.
+            # The format of a model is defined by `struct bt_mesh_model` type.
+            # All types are declared in `zephyr/include/zephyr/bluetooth/mesh/access.h`.
+            cid, pid, vid, _, elems_count, elems_ptr = struct.unpack('HHHHII', cdp0_value)
+
+            comp.cid = cid
+            comp.pid = pid
+            comp.vid = vid
+
+            elems_value = self.__read_symbol_data(elf, elems_ptr)
+            elems_iter = struct.iter_unpack('IHBBII', elems_value)
+            i = 0
+
+            for elem in elems_iter:
+                i += 1
+                if i > elems_count:
+                    print("Extracted more elems than \'elem_count\'")
+                    sys.exit(1)
+                __rt, loc, sig_count, vnd_count, sig_ptr, vnd_ptr = elem
+                elem_item = comp.elem_add(loc)
+
+                def models_unpack(ptr, elem_item, vnd):
+                    models_value = self.__read_symbol_data(elf, ptr)
+                    model_format = 'HHIIIHHIHH' + ('I' if label_cnt > 0  else '') + 'II' + ('I' if lcd_srv else '')
+                    models_iter = struct.iter_unpack(model_format, models_value)
+
+                    for model in models_iter:
+                        id1, id2, __rt, __pub, __keys, __keys_cnt, _, __groups, __groups_cnt, _, __uuids, __op, __cb = model
+                        if not vnd:
+                            elem_item.sig_model_add(id1)
+                        else:
+                            elem_item.vnd_model_add(id1, id2)
+
+                if sig_count > 0:
+                    models_unpack(sig_ptr, elem_item, False)
+                if vnd_count > 0:
+                    models_unpack(vnd_ptr, elem_item, True)
+
+    def extract_comp_data(self, kconfigs, comp):
+        """
+        Extract composition data from .elf file.
+
+        Parameters:
+            kconfigs (dict): A dictionary representing Kconfig options used for the firmware to compile with
+            comp (dict): A Composition Data dictionary class object to be filled in
+
+        Returns:
+            TBA
+        """
+        for addr in self.addrs:
+            self.__read_comp_data(addr, kconfigs, comp)
+
+
+def parse_kconfig(config_path):
+    """
+    Extracts content of '.config' file into a dictionary
+
+    Parameters:
+        config_path (str): Path to '.config' file used by a firmware
+    Returns:
+        dict: A dictionary where key is Kconfig option name and value is the value of the corresponding Kconfig option.
+    """
+    with open(config_path, 'r') as config:
+        configs = {}
+        for line in config:
+            if not line.startswith("CONFIG_"):
```

Comment:
Could this be CONFIG_BT_MESH_? To reduce the number of extracted sets.

Reply:
It is not only BT_MESH options that are extracted here.

Reply:
It has minimal impact on execution time with and without.